https://code.videolan.org/videolan/dav1d/-/issues/370
https://code.videolan.org/videolan/dav1d/-/commit/c6a08b3aa1ee99dade53e5e32033bc1d14455a22

From c6a08b3aa1ee99dade53e5e32033bc1d14455a22 Mon Sep 17 00:00:00 2001
From: Janne Grunau <janne-vlc@jannau.net>
Date: Tue, 21 Sep 2021 09:30:14 +0200
Subject: [PATCH 1/5] build: avoid meson's symbols_have_underscore_prefix

Meson's dynamic check is unreliable when additional compiler flags are
passed via CFLAGS. For example '-fprofile-instr-generate' in oss-fuzz'
coverage build. Fixes #370.
---
 meson.build | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1bf69ab..1a7c409 100644
--- a/meson.build
+++ b/meson.build
@@ -382,7 +382,11 @@ endif
 
 cdata.set10('ARCH_PPC64LE', host_machine.cpu() == 'ppc64le')
 
-if cc.symbols_have_underscore_prefix()
+# meson's cc.symbols_have_underscore_prefix() is unfortunately unrelieably
+# when additional flags like '-fprofile-instr-generate' are passed via CFLAGS
+# see following meson issue https://github.com/mesonbuild/meson/issues/5482
+if (host_machine.system() == 'darwin' or
+   (host_machine.system() == 'windows' and host_machine.cpu_family() == 'x86'))
     cdata.set10('PREFIX', true)
     cdata_asm.set10('PREFIX', true)
 endif
-- 
2.32.0

